modules/sagemaker/sagemaker-templates-service-catalog/templates/model_deploy/product_stack.py

```diff
@@ -110,102 +110,127 @@ def __init__(
         Tags.of(self).add("sagemaker:project-id", sagemaker_project_id)
         Tags.of(self).add("sagemaker:project-name", sagemaker_project_name)
 
+        dev_account_id: str = Aws.ACCOUNT_ID
+        dev_region: str = Aws.REGION
+        pre_prod_account_id = Aws.ACCOUNT_ID if not pre_prod_account_id else pre_prod_account_id
+        prod_account_id = Aws.ACCOUNT_ID if not prod_account_id else prod_account_id
+        pre_prod_region = Aws.REGION if not pre_prod_region else pre_prod_region
+        prod_region = Aws.REGION if not prod_region else prod_region
```

Comment:
Why do we need these lines? These 4 values should already be set with defaults when they're resolved as CfnParameters.

Reply:
These values for preprod and prod can still be removed. I don't think they're actually doing anything. The resolution of the `if` statement happens at synth time in CDK, whereas CFNParameters are resolved at deploy time. So a synth time, variables like `prod_account_id` are never going to be treated as None.